core/registry.js

```diff
@@ -167,8 +174,33 @@ Blockly.registry.getClass = function(type, name) {
 };
 
 /**
- * Gets the class name or the class from Blockly options for the given type.
- * This is used for plugins that override a built in feature. (Ex: Toolbox)
+ * Gets the class for the given name and type.
+ * @param {string|Blockly.registry.Type<T>} type The type of the plugin.
+ *     (ex: Field, Renderer)
+ * @param {string} name The plugin's name. (Ex. field_angle, geras)
+ * @return {?function(new:T, ...?)} The class with the given name and type or
+ *     null if none exists.
+ * @template T
+ */
+Blockly.registry.getClass = function(type, name) {
+  return /** @type {?function(new:T, ...?)} */ (Blockly.registry.getRegistryItem_(type, name));
+};
+
+/**
+ * Gets the object for the given name and type.
+ * @param {string|Blockly.registry.Type<T>} type The type of the plugin.
+ *     (ex: Category)
+ * @param {string} name The plugin's name. (Ex. logic_category)
+ * @returns {T} The object with the given name and type or null if none exists.
+ * @template T
+ */
+Blockly.registry.getObject = function(type, name) {
```

Comment:
Just wanting to get an idea of what other objects we have in mind here?

Ones I can think of: 
- Blocks
- Block generators
- Generators
- Themes

I can see a solid use case for Themes, and maybe Generators, but unclear whether we would want to change the pattern for registering blocks / block generators (functions).